eng/scripts/Remove-WormStorageAccounts.ps1

```diff
@@ -0,0 +1,38 @@
+# Helper script for removing storage accounts with WORM that sometimes get leaked from live tests not set up to clean
+# up their resource policies
+
+[CmdletBinding(SupportsShouldProcess=$True)]
+param(
+    [string]$GroupPrefix
+)
+
+# Be a little defensive so we don't delete non-live test groups via naming convention
+if (!$groupPrefix -or !$GroupPrefix.StartsWith('rg-')) {
+    Write-Error "The -GroupPrefix parameter must start with 'rg-'"
+    exit 1
+}
+
+$groups = Get-AzResourceGroup | ? { $_.ResourceGroupName.StartsWith($GroupPrefix) } | ? { $_.ProvisioningState -ne 'Deleting' }
+
+foreach ($group in $groups) {
+    $accounts = Get-AzStorageAccount -ResourceGroupName $group.ResourceGroupName
+    if ($accounts) {
+        foreach ($account in $accounts) {
+            if ($PSCmdlet.ShouldProcess($group.ResourceGroupName)) {
+                Write-Host "Removing $($account.StorageAccountName) in $($account.ResourceGroupName)"
+            } else {
+                Write-Host "What if: Removing $($account.StorageAccountName) in $($account.ResourceGroupName)"
```

Comment:
I thought the whatif logic in shouddprocess would already output this

Reply:
Did you want to ship if whatif is set? I see you are still calling remove below.

Reply:
I fixed it up to just check `$WhatIfPreference`. The extra stuff from the `ShouldProcess` log makes the output hard to parse in my opinion.

I still call Remove below because the parent `-WhatIf` will still pass through to the Az calls and I want those to be part of the whatif logging since it shows each container that would be deleted.